plugin/trino-hive/src/main/java/io/trino/plugin/hive/projection/DateProjection.java

```diff
@@ -73,12 +75,32 @@ final class DateProjection
     private static final Pattern DATE_RANGE_BOUND_EXPRESSION_PATTERN = Pattern.compile("^\\s*NOW\\s*(([+-])\\s*([0-9]+)\\s*(DAY|HOUR|MINUTE|SECOND)S?\\s*)?$");
 
     private final String columnName;
+    private final String dateFormatPattern;
     private final DateTimeFormatter dateFormat;
     private final Supplier<Instant> leftBound;
     private final Supplier<Instant> rightBound;
     private final int interval;
     private final ChronoUnit intervalUnit;
 
+    @JsonCreator
+    public DateProjection(
+            @JsonProperty("columnName") String columnName,
+            @JsonProperty("dateFormatPattern") String dateFormatPattern,
+            @JsonProperty("leftBound") Instant leftBound,
+            @JsonProperty("rightBound") Instant rightBound,
+            @JsonProperty("interval") int interval,
+            @JsonProperty("intervalUnit") ChronoUnit intervalUnit)
+    {
+        this.columnName = requireNonNull(columnName, "columnName is null");
+
+        this.dateFormatPattern = requireNonNull(dateFormatPattern, "dateFormat is null");
+        this.dateFormat = DateTimeFormatter.ofPattern(dateFormatPattern);
+        this.leftBound = () -> leftBound;
+        this.rightBound = () -> rightBound;
```

Comment:
Wrapping these with `Supplier` seems weird, and it creates some ambiguity around whether `DateProjection` is actually serializable in a safe way. The only mechanism that seems to rely on this is `DateExpressionBound` which produces a different `Instant.now()` on each access- but I don't see why that should be necessary. Maybe this can be simplified out?

Reply:
I think can. 
And I feels that maybe we need to refactor the `DateProjection` a bit before apply the read/write fix, WDYT? And this refactor probably should go into the first commit